Delivers [#158795315  #158795461] User should be able to sign in and out

LarryKarani opened the pull request:
What does this PR do?
Adds signup endpoint to the API

Description of Task to be completed
Adding a signup and login endpoint to allow users to sign up

POST /api/v1/signup
POST/api/v1/login

Tasks achieved in this PR

POST /api/v1/signup
POST/api/v1/login
How should one test this endpoint?

Clone this repo using git clone 
Check the ReadMe for instructions on setting up the project.
git checkout f1
Run pytest ridemyway/tests/test_v2/test_auth/test_signup.py to see the signup tests pass. Alternatively, 


**Background information**
The build fails because its a testing endpoints which are yet to be implemented. 

Resources & Libs
This Pivotal Tracker board was used to plan and manage the tasks.

Relevant PIvotal Tracker Stories?
#158795315 #158795461

LarryKarani committed:
added procfile and modified routes

LarryKarani committed:
added procfile and modified routes

LarryKarani committed:
added procfile and modified routes

LarryKarani committed:
edited the procfile

LarryKarani committed:
add gunicorn on requirements.txt file

LarryKarani committed:
edited the procfile

LarryKarani committed:
edited the procfile

LarryKarani committed:
edited removed .yml

LarryKarani committed:
removed unused variable

LarryKarani committed:
removed unused variable

LarryKarani committed:
removed unused jwt imports

LarryKarani committed:
Connect db to the ride mywayapp
updated routes to queery the db
updated the signup feature finishesh[#15895315]

LarryKarani committed:
add login fuctionality with jwt finishes[#158795461]

LarryKarani committed:
add travis to f1 branch
prevent same user registration

JoshuaOndieki commented:
Consider debugging your app a bit. The `ExpiredSignatureImport` Error that was occurring on your Heroku builds has been ported into travis. YOu build is failing not just because you haven't implemented the other features but also because you have import errors which shouldn't be there, I guess.

JoshuaOndieki commented on FlaskApp/api/app.py:
```diff
@@ -3,22 +3,47 @@
+
+    create_table(con, user_table_queries)
+    create_table(con, ride_offer_queries)
+    create_table(con, request_queries)
```
If you can manage, I'd suggest you separate database migration and app factory. I.e. maybe it would be a good idea not to migrate your DB everytime your app is run. My understanding of data persistence is that your data remains even if your server is restarted a billion times, your server doesn't have to reset the DB everytime as it is in the in-memory situation

JoshuaOndieki commented on FlaskApp/api/routes.py:
```diff
@@ -64,26 +65,29 @@ def post(self):
 
-        new_user = User(data['name'].strip(), data['username'].strip(), data['email'].strip().lower(),generate_password_hash(data['password'].strip())).register_user()
+        new_user_query = 'INSERT INTO users (username, email, password)\
+         VALUES(\'%s\',\'%s\',\'%s\');' %(data['username'].strip(),data['email'].strip().lower(),\
```
SQL injection possible here.
Have a look at this :arrow_right: https://security.openstack.org/guidelines/dg_parameterize-database-queries.html

LarryKarani replied:
Thanks for  the feedback realy helpful

xcixor commented on FlaskApp/api/db_config.py:
```diff
@@ -0,0 +1,20 @@
+    print (var)
+
+except Exception as e:
+    print("Error %d: %s" % (e.args[0], e.args[1]))
```
You should return this error instead of printing it

xcixor commented on Procfile:
```diff
@@ -0,0 +1 @@
+web: gunicorn manage:app
```
new line here

LarryKarani replied:
Thanks ill do that

kwameasiago commented on .travis.yml:
```diff
@@ -1,22 +1,20 @@
\ No newline at end of file
+after_success:
+  -coveralls
+  -codeclimate-test-reporter
```
Add a line at the end of the file

LarryKarani replied:
Thank you

LarryKarani committed:
Update app to only query from db
- deleted unused files(request.py, user.py)
-updated tests to cover api queries

LarryKarani committed:
unstalled jwt to make the build work

LarryKarani committed:
Remove white spaces
add .coveragerc ignore file
